libqtile/backend/x11/window.py

```diff
@@ -1044,14 +1044,15 @@ def change_layer(self, up=True, top_bottom=False):
         windows.sort(key=lambda w: stack.index(w[0].wid))
 
         # Get lists of windows on lower, higher or same "layer" as window
-        lower = [w[0].wid for w in windows if w[1] is None and w[2] > layering]
-        higher = [w[0].wid for w in windows if w[1] is None and w[2] < layering]
-        same = [w[0].wid for w in windows if w[1] is None and w[2] == layering]
+        lower = [w[0].wid for w in windows if w[2] > layering]
+        higher = [w[0].wid for w in windows if w[2] < layering]
+        same = [w[0].wid for w in windows if w[2] == layering]
 
         # We now need to identify the new position in the stack
 
         # If the window has a parent, the window should just be put above it
-        if parent:
+        # If the parent isn't being managed by qtile then it may not be stacked correctly
+        if parent and parent in self.qtile.windows_map:
```

Comment:
I've no idea why Google Earth says that some dialogs are transient windows for a window that we're not managing (and not displaying). Is there a better way to handle that scenario>

Reply:
If we aren't managing the parent window, could we ignore the transient hint and treat it as a normal window?

Reply:
That's essentially what that line does. i.e. if we're not managing it, this condition isn't met and we'll fall back to normal rules.